File: dmpg18g1.py
from collections import deque

num_chairs, empty_chairs = map(int, input().split())
chairs = [None for i in range(num_chairs)]

for i in map(int, input().split()):
	chairs[i-1] = 0

full_chairs = list(map(int, input().split()))
for i in range(len(full_chairs)):
	chairs[full_chairs[i]-1] = i+1 if chairs[full_chairs[i]-1] is None else None # one-indexed input can go die in a fire

stack = deque()
while True:
	for i in range(len(chairs)):
		if chairs[i] == 0:
			if len(stack) > 0:
				chairs[i] = None
				stack.pop()
		if chairs[i] is not None and chairs[i] != 0:
			stack.append(chairs[i])
			chairs[i] = None
	if len(stack) == 1:
		break
print(stack.pop())

File: dmpg18g1.cpp
#include <iostream>
#include <stack>

int chairs[1000001];
int main() {
	int numChairs, emptyChairs; std::cin >> numChairs >> emptyChairs;
	
	for (int i = 0; i < emptyChairs; i++) {
		int x; std::cin >> x;
		chairs[x-1] = -1;
	}

	for (int i = 0; i < emptyChairs + 1; i++) {
		int index; std::cin >> index;
		chairs[index-1] = chairs[index-1] == 0 ? i+1 : 0;
	}

	std::stack<int> stack;
	while (true) {
		for (int i = 0; i < numChairs; i++) {
			if (chairs[i] == -1) {
				if (stack.size() > 0) {
					chairs[i] = 0;
					stack.pop();
				}
			}
			if (chairs[i] != 0 && chairs[i] != -1) {
				stack.push(chairs[i]);
				chairs[i] = 0;
			}
		}

		if (stack.size() == 1) break;
	}

	std::cout << stack.top() << '\n';
	return 0;
}